### PR TITLE
add getformat

### PR DESCRIPTION
**Pull request by Miragecore**

### CODE REVIEW

File: vm/src/builtins/float.rs [hunk 1]
```diff
@@ -204,6 +204,24 @@ impl PyFloat {
         }
     }
 
+    #[pymethod(magic)]
+    fn getformat(&self, spec: PyStrRef, vm: &VirtualMachine) -> PyResult<String> {
+        if !matches!(spec.as_str(), "double" | "float") {
+            return Err(vm.new_value_error(format!(
```
Comment (youknowone) [inline vm/src/builtins/float.rs:210]: Please check this clippy advice

**Review by youknowone — CHANGES_REQUESTED** [review]

**Comment by Miragecore** [comment]
Because of this issue https://github.com/rust-lang/rust-clippy/issues/8527,
I can't check this on my local.

**Comment by Miragecore** [comment]
Because of this issue https://github.com/rust-lang/rust-clippy/issues/8527, I can't check this on my local.

**Review by youknowone — APPROVED** [review]
Thank you for contributing!

**Pull request merged**